python/pyspark/sql/functions.py

```diff
@@ -172,7 +173,9 @@ def lit(col: Any) -> Column:
         return col
     elif isinstance(col, list):
         if any(isinstance(c, Column) for c in col):
-            raise ValueError("lit does not allow a column in a list")
+            raise PySparkException(
+                error_class="COLUMN_IN_LIST", message_parameters={"funcName": "lit"}
+            )
```

Comment:
ditto.

Will migrate more errors in large batches as follow-ups

I believe it would be a big task, so I will create several sub-tasks under [umbrella JIRA](https://issues.apache.org/jira/browse/SPARK-41597).